qtfred/src/mission/dialogs/CampaignEditorDialogModel.cpp

```diff
@@ -0,0 +1,801 @@
+#include "CampaignEditorDialogModel.h"
+#include "weapon/weapon.h"
+#include "cutscene/cutscenes.h"
+#include "menuui/mainhallmenu.h"
+#include "stats/scoring.h"
+#include "mission/missiongoals.h"
+#include "mission/missionsave.h"
+
+#include <QPlainTextDocumentLayout>
+
+extern int Skip_packfile_search;
+
+extern void parse_init(bool basic);
+extern void parse_mission_info(mission *mn, bool basic);
+extern void parse_events(mission *mn);
+extern void parse_goals(mission* mn);
+extern int Subsys_status_size;
+
+namespace fso {
+namespace fred {
+namespace dialogs {
+
+static QString loadFile(QString file, const QString& campaignType) {
+	if (file.isEmpty()) {
+		mission_campaign_clear();
+		Campaign.type = campaignType.isEmpty() ? CAMPAIGN_TYPE_SINGLE :
+												 CampaignEditorDialogModel::campaignTypes.indexOf(campaignType);
+		return {};
+	}
+	//FRED is to enforce that only on new campaigns a campaign type may be given
+	Assert(campaignType.isEmpty());
+	parse_init(false);
+	if (mission_campaign_load(qPrintable(file.replace('/',DIR_SEPARATOR_CHAR)), nullptr, 0))
+		return {};
+
+	return Campaign.filename;
+}
+
+static void initShips(const SCP_vector<ship_info>::const_iterator &s_it, CheckedDataListModel<std::ptrdiff_t> &model){
+	std::ptrdiff_t shpIdx{ std::distance(Ship_info.cbegin(), s_it) };
+	if (s_it->flags[Ship::Info_Flags::Player_ship]) {
+		model.initRow(
+				s_it->name,
+				shpIdx,
+				static_cast<bool>(Campaign.ships_allowed[static_cast<size_t>(shpIdx)]));
+}
+}
+
+static void initWeps(const SCP_vector<weapon_info>::const_iterator &w_it, CheckedDataListModel<std::ptrdiff_t> &model){
+	std::ptrdiff_t wepIdx{ std::distance(Weapon_info.cbegin(), w_it) };
+	for (const ship_info& shp: Ship_info) {
+		if (shp.flags[Ship::Info_Flags::Player_ship]
+			&& shp.allowed_weapons[static_cast<size_t>(wepIdx)]
+			&& !model.contains(w_it->name)) {
+			model.initRow(
+					w_it->name,
+					wepIdx,
+					static_cast<bool>(Campaign.weapons_allowed[static_cast<size_t>(wepIdx)]));
+}
```

Comment:
Style inconsistency.  There are no comments here stating which brace ends which namespace scope.

Reply:
This anonymous namespace ends after `parseMnPart`, as commented.